UI.PageModels/Pages/Admin/Booking/ParticipantsPage.cs

```diff
@@ -158,6 +187,26 @@ private void ClickAddParticipantAndWait()
             500); // THIS IS ABSOLUTELY REQUIRED - the component takes 500ms to respond to change based on a setTimeout
     }
 
+    private void ClickUpdateParticipantAndWait()
```

Comment:
the methods `ClickUpdateParticipantAndWait` and `ClickConfirmRemoveParticipantAndWait` are not re-used. I would suggest to move the code into the calling method so the actions to `UpdateParticipant` and `RemoveParticipant` are complete. Alternatively create a re-useable method that accepts 1 param

- elementToClick

```
private void ClickElementAndWaitToDisappear(By element)
     {
         Thread.Sleep(
             500); // THIS IS ABSOLUTELY REQUIRED - the component takes 500ms to respond to change based on a setTimeout
         ClickElement(element);
         WaitForElementToBeInvisible(element, 5);
         Thread.Sleep(
             500); // THIS IS ABSOLUTELY REQUIRED - the component takes 500ms to respond to change based on a setTimeout
     }
```

so the private method becomes re-usable as:

```
public void UpdateParticipant(string fullName, string newDisplayName)
{
   var editLink = GetEditLink(fullName);
   ClickElement(editLink);
   if (_useParty)
   {
       WaitForDropdownListToPopulate(_partyDropdown);
   }
   WaitForDropdownListToPopulate(_roleDropdown);
   EnterText(_displayNameTextfield, newDisplayName);

   ClickElementAndWaitToDisappear(_updateParticipantButton);
}
```

and

```
public void RemoveParticipant(string fullName)
{
   var removeLink = GetRemoveLink(fullName);
   ClickElement(removeLink);

   ClickElementAndWaitToDisappear(_confirmRemoveParticipantButton);
}
```

Reply:
good idea, thanks. Updated